src/lib/moderation/managers/ModerationManager.ts

```diff
@@ -173,7 +171,7 @@ export class ModerationManager extends Collection<number, ModerationEntity> {
 			})
 		);
 
-		return lock.resolve;
+		return () => lock.resolve();
```

Comment:
can't we use `lazy` from @sapphire/utilities here? Or will that be an issue because it caches?
```ts
/**
 * Lazily creates a constant or load a module and caches it internally
 * @param cb The callback to lazily run
 * @returns The value returned by the callback, or the cached value if it was already initialised once.
 */
export function lazy<T>(cb: () => T) {
	let defaultValue: T;

	return () => (defaultValue ??= cb());
}

```

Reply:
That's not what the code does.